pkg/perf/iperf_task.go

```diff
@@ -104,35 +105,46 @@ func (t *IperfTest) Run(ctx context.Context) (interface{}, error) {
 }
 
 func (t *IperfTest) runIperfTest(ctx context.Context, clientIP string, tcp bool) IperfResult {
-	iperfClient := goIperf.NewClient(clientIP)
-	iperfClient.SetBandwidth("1M")
-	iperfClient.SetPort(iperf.IperfPort)
-	iperfClient.SetInterval(20)
-	iperfClient.SetJSON(true)
+	opts := make([]string, 0)
+	opts = append(opts,
+		"--client", clientIP,
+		"--bandwidth", "1M",
+		"--port", fmt.Sprint(iperf.IperfPort),
+		"--interval", "20",
+		"--json",
+	)
 
 	if !tcp {
-		iperfClient.SetLength("16B")
-		iperfClient.SetProto(goIperf.PROTO_UDP)
+		opts = append(opts, "--length", "16B", "--udp")
 	}
-
-	err := iperfClient.Start()
-	if err != nil {
-		log.Error().Err(err).Msgf("failed to start iperf client with ip '%s'", clientIP)
+	output, err := exec.CommandContext(ctx, "iperf", opts...).CombinedOutput()
+	exitErr := &exec.ExitError{}
+	if err != nil && !errors.As(err, &exitErr) {
```

Comment:
Actually I'm not sure I follow, we ignore errors other than exitErr? while that actually means we failed to execute the binary?

Reply:
we ignore exit errors because `iperf` exits with a non-zero code in case it failed to run and prints the error reason in `error` field in JSON output. For example:

```json
{
  "start": {
    "connected": [],
    "version": "iperf 3.14",
    "system_info": "Linux zero-os 5.10.55-Zero-OS #2 SMP PREEMPT Wed Aug 9 13:31:25 UTC 2023 x86_64"
  },
  "intervals": [],
  "end": {},
  "error": "unable to connect to server - server may have stopped running or use a different port, firewall issue, etc.: Network unreachable"
}
```